src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -4,16 +4,19 @@
  */
 
 import { Client } from '@opensearch-project/opensearch';
+import { Credentials } from 'aws-sdk';
```

Comment:
`Credentials` is marked as deprecated in `aws-sdk/types@3`. Are we okay with using this?

Reply:
I believe it's OK if we are using the aws-sdk@v2 approach to use sigv4 in opensearch-js client. Both v2, v3 approaches are supported by client

Reply:
V2 has `AWS.Credential` class, while V3 deprecated it and load credential from sources by chain https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html. In MD sigV4 use case, the credentials are inputs from user, not from env, shared file or any sources in V3. 
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_credential_providers.html#f

Reply:
V2 has `AWS.Credential` class, while V3 deprecated it and load credential from sources by chain https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html. In MD sigV4 use case, the credentials are inputs from user, not from env, shared file or any sources in V3. 
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_credential_providers.html#f